src/Concordium/Client/GRPC2.hs

```diff
@@ -2552,20 +2854,255 @@ getAccountInfoV2 ::
+        -- This is here so we may leverage `withGRPCCoreV2`.
+        callHelper client = do
+            res <- rawUnary method client input
+            return $ fmap RawUnaryOutput res
+
+-- |Call a streaming V2 GRPC API endpoint and return the (collected) results in a sequence.
+-- Note that some care should be taken when using this with long-running calls. See
+-- `withGRPCCoreV2` for more info.
+withServerStreamCollectV2 ::
+    ( HasMethod CS.Queries m,
+      MonadIO n,
+      i ~ MethodInput CS.Queries m,
+      o ~ Seq.Seq (MethodOutput CS.Queries m) ) =>
+    -- |The procedure to call.
+    RPC CS.Queries m ->
+    -- |The procedure input.
+    i ->
+    -- |A mapping of the collected result.
+    (GRPCResult o -> b) ->
+    ClientMonad n b
+withServerStreamCollectV2 method input =
+    withServerStreamCallbackV2 method input mempty handler
+    where handler acc o = return $ acc <> pure o
+
+-- |Call a streaming GRPC API endpoint and return the (collected) results.
+-- Takes a `fold`-like callback and an accumulator, and returns the result
+-- of folding through each stream element, once the stream terminates. Note
+-- that some care should be taken when using this with long-running calls.
+-- See `withGRPCCoreV2` for more info.
+withServerStreamCallbackV2 ::
+    ( HasMethod CS.Queries m,
+      MonadIO n,
+      i ~ MethodInput CS.Queries m,
+      o ~ MethodOutput CS.Queries m ) =>
+    -- |The procedure to call.
+    RPC CS.Queries m ->
+    -- |The procedure input.
+    i ->
+    -- |An initial `fold`-like accumulator that
+    -- is updated with the handler each time a
+    -- stream element arrives.
+    a ->
+    -- |A `fold`-like handler which is used
+    -- to process a stream object and update
+    -- the accumulator.
+    (a -> o -> ClientIO a) ->
+    -- |A mapping of the accumulated result.
+    (GRPCResult a -> b) ->
+    ClientMonad n b
+withServerStreamCallbackV2 method input acc handler k =
+    withGRPCCoreV2 callHelper k
+    where
+        -- This is simply a handler which conforms to the one required by
+        -- `rawStreamServer`. This is here, so we may ignore the response
+        -- headers in the handler, just for the sake of ergonomics.
+        handler' acc' _hds streamObj = handler acc' streamObj
+        -- Helper that invokes the streaming call with the above callback.
+        -- This is here so we may leverage `withGRPCCoreV2`.
+        callHelper client = do
+            res <- rawStreamServer method client acc input handler'
+            return $ fmap ServerStreamOutput res
+
+-- |Run a request helper function with a client instance to call a GRPC procedure.
+-- The output is interpreted using the function given in the second parameter.
+--
+-- If the client fails during a call, the connection will be re-established at most
+-- a number of of times depending on the configuration. It should, however, be noted
+-- that for performance reasons, the client is a (global) instance protected by a
```

Comment:
What do you mean "for performance reasons"? Performance is not the reason for using the RWLock.

the reason for using it is so that reconnects can be handled since the library we have is shitty and does not properly handle timeouts, among other things.

Reply:
> Performance is not the reason for using the RWLock.

Right. But the message I am trying to convey in that sentence is not about the RWLock, but rather about the single (global) client instance, i.e.:

> for performance reasons, _the client is a (global) instance_

If I am wrong about this, or if this is poorly worded, then the reason should be rephrased or perhaps completely omitted.

Reply:
It's not really a global instance in the normal sense of the word. You are meant to use the same connection for multiple requests since establishing connections is expensive. You should write that since these comments about "global" instance are misleading. As in, I wrote this setup in the client, and I don't understand what you mean by the comment, so I think other readers will likely not either.